packages/rich-text/src/component/use-format-boundaries.js

```diff
@@ -12,133 +12,125 @@ import { isCollapsed } from '../is-collapsed';
 
 const EMPTY_ACTIVE_FORMATS = [];
 
-export function useFormatBoundaries( {
-	record,
-	applyRecord,
-	setActiveFormats,
-} ) {
-	return useRefEffect(
-		( element ) => {
-			function onKeyDown( event ) {
-				const { keyCode, shiftKey, altKey, metaKey, ctrlKey } = event;
+export function useFormatBoundaries( props ) {
+	const propsRef = useRef( props );
+	propsRef.current = propsRef;
```

Comment:
This avoids having to remove and re-add event handlers every time the selection changes.